File: Max Number of K-Sum Pairs.js
var maxOperations = function(nums, k) {
    
    let frequencyCounter ={};
    let counter = 0
    
    for(let val of nums){
        frequencyCounter[val] =(frequencyCounter[val] || 0) +1;
    }
    
    for(let val of nums){
        let complement = k - val;
        if(frequencyCounter[val] === 0){
            continue;
        }
        frequencyCounter[val] -=1
        if(frequencyCounter[complement] >= 1){
            frequencyCounter[complement] -=1;
            counter++
            
        }
    }
    
    return counter;
   
    
};

File: Max Number of K-Sum Pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        count = defaultdict(int)
        result = 0
        for num in nums:
            if (k - num) in count and count[k-num] > 0:
                result += 1
                count[k-num] -= 1
            else:
                count[num] += 1
        return result
